nemo/collections/asr/parts/k2/graph_decoders.py

```diff
@@ -0,0 +1,210 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, Optional, Tuple, Union
+
+import k2
+import torch
+
+from nemo.collections.asr.parts.k2.topologies import build_topo
+from nemo.collections.asr.parts.k2.utils import (
+    create_supervision,
+    intersect_with_self_loops,
+    invert_permutation,
+    load_graph,
+    make_blank_first,
+    prep_padded_densefsavec,
+    shift_labels_inpl,
+)
+from nemo.utils import logging
+
+
+class DecoderConf:
```

Comment:
Might as well make it a dataclass (suggestion).
Avoid using **kwargs at all costs.

Reply:
Done in [2045660](https://github.com/NVIDIA/NeMo/pull/3351/commits/2045660b6cab2d45d3c8d3c018d2ebe9e05f1faa)